merge/update.go

```diff
@@ -179,7 +179,7 @@ func (s *Updater) Update(liveObject, newObject *typed.TypedValue, version fieldp
 		ignored = fieldpath.NewSet()
 	}
 	managers[manager] = fieldpath.NewVersionedSet(
-		managers[manager].Set().Union(compare.Modified).Union(compare.Added).Difference(compare.Removed).RecursiveDifference(ignored),
+		managers[manager].Set().Difference(compare.Removed).Union(compare.Modified).Union(compare.Added).RecursiveDifference(ignored),
```

Comment:
is this the only functional line changing?

I'm trying to figure out why the order change matters

Reply:
When we go from duplicate to non-duplicate and vice versa, we add/remove the managedfields for duplicates and then remove/add the managedfields for the non-duplicated form. That does an add AND a remove, possibly of the same key. Because of how the line was written before, we would add the add and then remove the remove, which doesn't work for fields that are in both.